localstack/services/apigateway/provider.py

```diff
@@ -242,9 +243,10 @@ def create_resource(
         store = get_apigateway_store(account_id=context.account_id, region=context.region)
 
         # save children to allow easy deletion of all children if we delete a parent route
-        rest_api_resources = store.resources_children.setdefault(rest_api_id, {})
-        children = rest_api_resources.setdefault(parent_id, [])
-        children.append(response["id"])
+        # FIXME: delete conditional before v2
+        if rest_api := store.rest_apis.get(rest_api_id):
```

Comment:
my initial comment as also related to abstract the store operations like this. Not sure if possible, but pretty much abstract the internals of the store into methods that can be refactored further without changing callers code in the future.

Reply:
It's a bit difficult, as the error message will be different depending on the operation. It's hard to abstract that away.
I've dealt with a provider implementing a lot of logic into the store, and it made it really hard to maintain, because after a while you need to implement operation specific behaviour into the store logic.